src/mesh/mesh_communication.C

```diff
@@ -1001,9 +1001,6 @@ void MeshCommunication::gather (const processor_id_type, DistributedMesh &) cons
 // ------------------------------------------------------------
 void MeshCommunication::gather (const processor_id_type root_id, DistributedMesh & mesh) const
 {
-  // The mesh should know it's about to be serialized
-  libmesh_assert (mesh.is_serial());
```

Comment:
This isn't related to the failure I'm seeing.

That said: this assert is wrong in N_proc out of N_proc+1 cases, but the fix isn't to delete it entirely, the fix is to make it more specific, e.g.  ```libmesh_assert(root_id != DofObject::invalid_processor_id || mesh.is_serial())```